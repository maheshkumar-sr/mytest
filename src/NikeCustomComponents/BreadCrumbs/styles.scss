@import '../../constants/color.scss';
.breadCrumbContainer {
    .breadCrumb {
        line-height: 1;

        .breadCrumbsPath{
            color: $blue;
            font-size: 13px;
            font-family: "Helvetica Neue", Helvetica, Arial, sans-serif;
        }

        li:last-child a.breadCrumbsPath{
            color: #787878;
            font-size: 13px;
        }
    }
}
