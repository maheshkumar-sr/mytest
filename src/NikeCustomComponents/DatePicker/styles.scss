@import '../../constants/variables.scss';

.datePickerContainer {
    .keyboardDatePickerContainer {
        margin: 0;

        input {
            font-size: 13px;
        }
    }

    .datePickerAdornmentRoot {
        button {
            padding: 0;
        }
    }

    .datePickerInputBaseRoot {
        &::before {
            border-bottom: 1px solid $grey2;
        }
    }
}