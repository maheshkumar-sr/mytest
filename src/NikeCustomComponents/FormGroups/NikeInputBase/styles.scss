@import '../../../constants/variables.scss';

.inputBaseContainer {
    width: 100%;
    display: flex;

    .searchInputBaseInput {
        font-size: 12px;
        padding: 8px;
    }

    .searchInputBase {
        flex: 1;
    }

    .clearText {
        color: $blue;
        background-color: $white;
        font-size: 13px;
        font-family: "Helvetica Neue", Helvetica, Arial, sans-serif;
    }

    .clearText:focus {
        outline: none;
    }
}