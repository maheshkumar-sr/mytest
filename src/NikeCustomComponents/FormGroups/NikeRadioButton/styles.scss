.iconSize {
      font-size: 1rem;
}
