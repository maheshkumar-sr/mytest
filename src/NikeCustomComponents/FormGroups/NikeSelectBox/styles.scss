@import '../../../constants/variables.scss';

.selectBoxContainer {
    .formControlContainer {
        height: 100%;
        width: 100%;
        
        .filterSelectFiledContainer {
            background-color: transparent;
            font-size: 13px;
            height: 100%;

            &::before, &::after {
                border-bottom: 1px solid $grey2;
            }

            .searchSelectRoot {
                margin: 0;
            }

            .searchSelect {
                background-color: transparent;
            }

            .searchSelectIcon {
                color: black;
                width: 18px;
                height: 18px;
                top: calc(50% - 9px);
            }
        }

        .selectFieldContainer {
            background-color: $grey3;
            font-size: 13px;
            height: 100%;
            width: 120px;

            &::before, &::after {
                 border: none;
            }

            &:hover::before {
                border: none !important;
            }

            .searchSelectRoot {
                margin: 8px 0 8px 8px;
            }

            .searchSelect {
                background-color: transparent;
            }

            .searchSelectIcon {
                color: black;
                width: 18px;
                height: 18px;
                top: calc(50% - 9px);
            }
        }

        .errorSelect {
            .selectFieldContainer {
                color: red;
            }
        }
    }
}

.menuPropsPaper {
    width: 250px;
    margin-top: 12px;
    margin-left: -10px;
    border-radius: 0 !important;
}

.filterMenuPropsPaper {
    width: 250px;
    border-radius: 0 !important;
}

.menuPropsPaper, .filterMenuPropsPaper {
    .menuPropsList {
        padding: 20px 10px;

        .menuItemRoot {
            font-size: 13px;

            &:hover {
                background-color: $lightBlue;
            }
        }

        .menuItemSelected {
            background-color: $lightBlue;
            display:none;
          }
          .menuItemDisabled {
            display: none;
            }
     }
}