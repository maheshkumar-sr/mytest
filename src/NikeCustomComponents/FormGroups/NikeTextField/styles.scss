@import '../../../constants/variables.scss';

.textFieldContainer {    
    .textFieldRoot {        
        input {
            color:$primaryblack;
            font-size: 13px;
        }
    }

    .textFieldInputBaseRoot {
        &::before {
            border-bottom: 1px solid $grey2;
        }
    }
}
