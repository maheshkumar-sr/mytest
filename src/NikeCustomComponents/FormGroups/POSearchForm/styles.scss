@import '../../../constants/variables.scss';

.formPaperContainer {
    .poSeachText {
       color: $primaryblack; 
       font-weight: bold;
       margin-bottom: 25px;
    //    font-family: Helvetica Neue Medium;
    }
    .errorOutline{
        padding: 0;
        margin: 0;
        display: flex;
        width: 502px;
        height: 50px;
        border: 1px solid red;
        box-shadow: none;
        border-bottom-left-radius: 0;
        border-top-left-radius: 0;

    }

    .paperComponent {
        padding: 0;
        margin: 0;
        display: flex;
        width: 502px;
        height: 50px;
        border: 1px solid $grey2;
        box-shadow: none;
        border-bottom-left-radius: 0;
        border-top-left-radius: 0;
    }
    
    .searchIconButton {
        padding: 12px;
        color: $white;
        background-color: $primaryblack;
        min-height: 50px;
        top: -1px;
        border-radius: 0 4px 4px 0;

        &:hover {
            background-color: $primaryblack;
        }
    }

    .errorDisp {
        color: $error;
        font-size: 13px;
        font-family: "Helvetica Neue", Helvetica, Arial, sans-serif;
        padding: 5px;
    }
}
