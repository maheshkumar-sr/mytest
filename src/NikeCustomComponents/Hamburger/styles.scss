@import '../../constants/variables.scss';

.hambergerCollaspedContainer{
    background: $primaryblack 0% 0% no-repeat padding-box;
    box-shadow: 0px 2px 4px #00000029;
    opacity: 1;
    border-top: 1px solid $grey3;
    height: 100%;

    .hambergerCollapsedMenu {
        padding: 0;

        .hambergerCollapsedMenuItems {
            justify-content: center;
            padding: 22px 0;

            &:first-child {
                margin-top: 12px;
            }

            &:hover .hambergerCollapsedMenuIcon {
                transform: scale(1.5);
            }

            .hambergerCollapsedMenuIcon {
                justify-content: center;
                min-width: 18px;
            }
        }

        .hambergerCollapsedMenuItemsSelected {
            background-color: $primaryblack;

            &:hover .hambergerCollapsedMenuIcon {
                transform: scale(1.5);
            }
        }
    }
}

.hambergerExpandedContainer {
    .hambergerExpendedMenu {
        color: $grey2;

        .listSubheader {
            height: 61px;
            padding: 22px 30px 22px 20%;
            text-align: center;
            line-height: 1;
        }
        
        .listSubheaderAvatar {
            height: 19px;
            width: 49.81px;
            border-radius: 0;
            display: inline-flex;
        }

        .hambergerExpandedMenuItems {
            padding: 22px 30px 22px 20%;
            line-height: 1;

            &:hover {
                background-color: $grey1;
                color: $white;
            }

            .hambergerExpandedMenuIcon {
                justify-content: flex-start;
                min-width: 18px;
                margin-right: 17px;
            }

            .hambergerExpandedMenuItemText {
                margin: 0;
                white-space: nowrap;
                overflow: hidden;
                text-overflow: ellipsis;
            }

            .menuItemTextPrimary {
                line-height: 1;
                font-size: 14px;
            }
        }

        .hambergerExpandedMenuItemsSelected {
            color: $white;

            &:hover {
                background-color: $grey1;
                color: $white;
            }
        }

        .verticalDivider {
            margin-bottom: 12px;
            background-color: $grey1;
            box-shadow: none;
        }
    }
}

.verticalDivider {
    background-color: $grey3;
}

.drawerContainer {
    .drawerPaperClass {
        background: $primaryblack 0% 0% no-repeat padding-box;
        box-shadow: 0px 2px 4px #00000029;
        opacity: 1;
        padding: 0;
    }
}
