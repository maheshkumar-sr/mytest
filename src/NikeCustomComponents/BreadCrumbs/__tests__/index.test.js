import React from 'react';
import { mount } from 'enzyme';
import { StaticRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { MuiThemeProvider } from '@material-ui/core/styles';
import CustomBreadcrumbs from '../index';

describe('<CustomBreadcrumbs /> ', () => {
  let PROPS = {};

  const breadCrumbList = new Map([
    ['/', [{ linkTo: '/', label: 'Home' }]],

    ['/poProcessing', [{ linkTo: '/', label: 'Home' },
      { linkTo: '/poProcessing', label: 'P.O Processing' }]],

    ['/search', [{ linkTo: '/', label: 'Home' },
      { linkTo: '/poProcessing', label: 'P.O Processing' },
      { linkTo: '/search', label: 'P.O Search' }]]
  ]);
  
  beforeEach(() => {
    PROPS = {
      breadCrumbList,
      pathName: '/',
      staticContext: undefined,
      selectError: 'fkdjdf',
      filterContainer: 'sflkd',
      filterContainer: 'kjkk',
      isSelectNative: true
    };
  });

  test('render the component correctly', () => {
    const store = createStore(() => ({}));

    const closeDrawerSpy = jest.fn();
    const wrapper = mount(
      <Provider store={store}>
        <MuiThemeProvider>
          <StaticRouter context={{}}>
            <CustomBreadcrumbs {...PROPS} closeDrawer={closeDrawerSpy} />
          </StaticRouter>
        </MuiThemeProvider>
      </Provider>
    );
    wrapper.find('Link').at(0).simulate("click");
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});
