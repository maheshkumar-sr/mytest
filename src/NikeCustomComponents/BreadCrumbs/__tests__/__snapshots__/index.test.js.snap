// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomBreadcrumbs />  render the component correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <ThemeProvider>
    <StaticRouter
      context={Object {}}
    >
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        staticContext={Object {}}
      >
        <withRouter(CustomBreadcrumbs)
          breadCrumbList={
            Map {
              "/" => Array [
                Object {
                  "label": "Home",
                  "linkTo": "/",
                },
              ],
              "/poProcessing" => Array [
                Object {
                  "label": "Home",
                  "linkTo": "/",
                },
                Object {
                  "label": "P.O Processing",
                  "linkTo": "/poProcessing",
                },
              ],
              "/search" => Array [
                Object {
                  "label": "Home",
                  "linkTo": "/",
                },
                Object {
                  "label": "P.O Processing",
                  "linkTo": "/poProcessing",
                },
                Object {
                  "label": "P.O Search",
                  "linkTo": "/search",
                },
              ],
            }
          }
          closeDrawer={[MockFunction]}
          filterContainer="kjkk"
          isSelectNative={true}
          pathName="/"
          selectError="fkdjdf"
        >
          <CustomBreadcrumbs
            breadCrumbList={
              Map {
                "/" => Array [
                  Object {
                    "label": "Home",
                    "linkTo": "/",
                  },
                ],
                "/poProcessing" => Array [
                  Object {
                    "label": "Home",
                    "linkTo": "/",
                  },
                  Object {
                    "label": "P.O Processing",
                    "linkTo": "/poProcessing",
                  },
                ],
                "/search" => Array [
                  Object {
                    "label": "Home",
                    "linkTo": "/",
                  },
                  Object {
                    "label": "P.O Processing",
                    "linkTo": "/poProcessing",
                  },
                  Object {
                    "label": "P.O Search",
                    "linkTo": "/search",
                  },
                ],
              }
            }
            closeDrawer={[MockFunction]}
            filterContainer="kjkk"
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            isSelectNative={true}
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
            pathName="/"
            selectError="fkdjdf"
            staticContext={Object {}}
          >
            <div
              className="breadCrumbContainer"
            >
              <WithStyles(ForwardRef(Breadcrumbs))
                className="breadCrumb"
                separator={
                  <Memo(NavigateNextIcon)
                    fontSize="small"
                  />
                }
              >
                <ForwardRef(Breadcrumbs)
                  className="breadCrumb"
                  classes={
                    Object {
                      "li": "MuiBreadcrumbs-li",
                      "ol": "MuiBreadcrumbs-ol",
                      "root": "MuiBreadcrumbs-root",
                      "separator": "MuiBreadcrumbs-separator",
                    }
                  }
                  separator={
                    <Memo(NavigateNextIcon)
                      fontSize="small"
                    />
                  }
                >
                  <WithStyles(ForwardRef(Typography))
                    className="MuiBreadcrumbs-root breadCrumb"
                    color="textSecondary"
                    component="nav"
                  >
                    <ForwardRef(Typography)
                      className="MuiBreadcrumbs-root breadCrumb"
                      classes={
                        Object {
                          "alignCenter": "MuiTypography-alignCenter",
                          "alignJustify": "MuiTypography-alignJustify",
                          "alignLeft": "MuiTypography-alignLeft",
                          "alignRight": "MuiTypography-alignRight",
                          "body1": "MuiTypography-body1",
                          "body2": "MuiTypography-body2",
                          "button": "MuiTypography-button",
                          "caption": "MuiTypography-caption",
                          "colorError": "MuiTypography-colorError",
                          "colorInherit": "MuiTypography-colorInherit",
                          "colorPrimary": "MuiTypography-colorPrimary",
                          "colorSecondary": "MuiTypography-colorSecondary",
                          "colorTextPrimary": "MuiTypography-colorTextPrimary",
                          "colorTextSecondary": "MuiTypography-colorTextSecondary",
                          "displayBlock": "MuiTypography-displayBlock",
                          "displayInline": "MuiTypography-displayInline",
                          "gutterBottom": "MuiTypography-gutterBottom",
                          "h1": "MuiTypography-h1",
                          "h2": "MuiTypography-h2",
                          "h3": "MuiTypography-h3",
                          "h4": "MuiTypography-h4",
                          "h5": "MuiTypography-h5",
                          "h6": "MuiTypography-h6",
                          "noWrap": "MuiTypography-noWrap",
                          "overline": "MuiTypography-overline",
                          "paragraph": "MuiTypography-paragraph",
                          "root": "MuiTypography-root",
                          "srOnly": "MuiTypography-srOnly",
                          "subtitle1": "MuiTypography-subtitle1",
                          "subtitle2": "MuiTypography-subtitle2",
                        }
                      }
                      color="textSecondary"
                      component="nav"
                    >
                      <nav
                        className="MuiTypography-root MuiBreadcrumbs-root breadCrumb MuiTypography-body1 MuiTypography-colorTextSecondary"
                      >
                        <ol
                          className="MuiBreadcrumbs-ol"
                        >
                          <li
                            className="MuiBreadcrumbs-li"
                            key="child-0"
                          >
                            <Link
                              className="breadCrumbsPath"
                              key=".$Home"
                              onClick={[Function]}
                              to="/"
                            >
                              <LinkAnchor
                                className="breadCrumbsPath"
                                href="/"
                                navigate={[Function]}
                                onClick={[Function]}
                              >
                                <a
                                  className="breadCrumbsPath"
                                  href="/"
                                  onClick={[Function]}
                                >
                                  Home
                                </a>
                              </LinkAnchor>
                            </Link>
                          </li>
                        </ol>
                      </nav>
                    </ForwardRef(Typography)>
                  </WithStyles(ForwardRef(Typography))>
                </ForwardRef(Breadcrumbs)>
              </WithStyles(ForwardRef(Breadcrumbs))>
            </div>
          </CustomBreadcrumbs>
        </withRouter(CustomBreadcrumbs)>
      </Router>
    </StaticRouter>
  </ThemeProvider>
</Provider>
`;
